lib/ansible/modules/network/ios/ios_facts.py

```diff
@@ -46,6 +46,7 @@
         not be collected.
     required: false
     default: '!config'
+    choices: ['all', 'hardware', 'config', 'interfaces']
```

Comment:
For the choices given here `!` can be prefixed to specify that a specific subset should not be collected (which is also mentioned in `gather_subset ` description). As `choices` is not updated in the spec as well [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/ios/ios_facts.py#L447). So I think this is not valid documentation.

Reply:
Are you suggesting I list all variations with '!' as well?

['all', 'hardware', 'config', 'interfaces', '!all', '!hardware', '!config', '!interfaces']